barretenberg/cpp/src/barretenberg/vm/avm/trace/bytecode_trace.cpp

```diff
@@ -26,41 +66,89 @@ std::vector<FF> AvmBytecodeTraceBuilder::pack_bytecode(const std::vector<uint8_t
     return contract_bytecode_fields;
 }
 
-void AvmBytecodeTraceBuilder::build_bytecode_columns()
+FF AvmBytecodeTraceBuilder::compute_public_bytecode_commitment(const std::vector<uint8_t>& contract_bytes)
+{
+    std::vector<FF> contract_bytecode_fields = encode_bytecode(contract_bytes);
+    FF running_hash = FF::zero();
+    for (auto& contract_bytecode_field : contract_bytecode_fields) {
+        running_hash = poseidon2::hash({ contract_bytecode_field, running_hash });
+    }
+    return running_hash;
+}
+
+size_t AvmBytecodeTraceBuilder::total_bytecode_length() const
+{
+    return std::accumulate(all_contracts_bytecode.begin(),
+                           all_contracts_bytecode.end(),
+                           static_cast<size_t>(0),
```

Comment:
I would've expected a 0 to just work but maybe not?

Reply:
Nah i think it complains about implicit int -> size_t